### PR TITLE
HOCS-4910: support somu-list multiple choices

### PR DESCRIPTION
**Pull request by LiamMacP**
The rendering of some contribution somu-list rows requires the loading
of choice list for business area as well as business unit. This supports
the loading of multiple lists.

This change also includes some refactoring to the somu-list to pull out
common methods into helper files.

### CODE REVIEW

File: src/shared/helpers/date-helpers.js [hunk 1]
```diff
@@ -9,6 +9,25 @@ function isDateTodayOrAfter(caseDeadline) {
...
+    const [year, month, day] = date.split('-');
+    return `${day}/${month}/${year}`;
+};
+
+const addDaysToDate = (dateString, days) => {
+    const date = new Date(dateString);
+    date.setDate(date.getDate() + days);
+    return date;
+};
+
```
Comment (robertdeniszczyc2) [inline src/shared/helpers/date-helpers.js:28]: Could add tests for these potentially

File: src/shared/helpers/object-helpers.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const getObjectNameValue = (object) => {
+    if (!object) { return ['', '']; }
+    return Object.entries(object)[0];
+};
+
+export {
+    getObjectNameValue
```
Comment (robertdeniszczyc2) [inline src/shared/helpers/object-helpers.js:7]: Could add tests for this potentially

**Review by robertdeniszczyc2 — COMMENTED** [review]
Just added a couple questions, any questions give me a shout,

Cheers,
Rob

File: server/services/form.js [hunk 3]
```diff
@@ -147,20 +147,30 @@ async function hydrateField(field, req) {
...
             field.props.somuType = somuTypeItem;
 
-            const somuItem = await req.listService.fetch('CASE_SOMU_ITEM', { ...req.params, somuTypeId: somuTypeItem.uuid });
-            field.props.somuItems = somuItem;
+            if (choices && typeof choices === 'object') {
+                const choiceObj = {};
+
+                for (const [name, choice] of Object.entries(choices)) {
+                    if (choice) {
+                        Object.assign(choiceObj, { [name]: await req.listService.fetch(choice, req.params) });
```
[the remark this answers is not in the document]
Comment (LiamMacP) [reply server/services/form.js:163]: Yes I expect there to be a couple within this choices - we should revisit if this becomes more messy.

File: src/shared/helpers/date-helpers.js (hunk 1 above)
Comment (LiamMacP) [reply src/shared/helpers/date-helpers.js]: Done

File: src/shared/helpers/object-helpers.js (hunk 2 above)
Comment (LiamMacP) [reply src/shared/helpers/object-helpers.js]: Done

File: src/shared/helpers/objectHelpers.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+/**
+ * @param object a object with a single value
+ * @returns {boolean} array with index 0 being the name and index 1 being the value
```
Comment (LiamMacP) [inline src/shared/helpers/objectHelpers.js:3]:
```suggestion
 * @returns {array} array with index 0 being the name and index 1 being the value
```

**Review by robertdeniszczyc2 — APPROVED** [review]

File: src/shared/common/forms/composite/somu-list.jsx [hunk 5]
```diff
@@ -22,112 +24,70 @@ const STATUS_OPTION = {
...
     }
 
-    renderContributionTable(somuItem) {
+    getChoicesValue(choice) {
         const { choices } = this.state;
+        const [name, value = ''] = getObjectNameValue(choice);
+
+        const choiceList = choices[name] ?? [];
+        return choiceList.find(item => item.value === value)?.label ?? value;
+    }
```
Comment (AlexW013) [inline src/shared/common/forms/composite/somu-list.jsx:60]: So this is the section that allows you to get the full title for the contribution?

Comment (LiamMacP) [reply src/shared/common/forms/composite/somu-list.jsx:60]: Yes - we pass in an object with key of the field and value of an array of choices.

**Pull request merged**